src/parser/SparqlParser.cpp

```diff
@@ -121,7 +125,46 @@ void SparqlParser::parseWhere(const string& str, ParsedQuery& query) {
   while (start < inner.size()) {
     size_t k = start;
     while (inner[k] == ' ' || inner[k] == '\t' || inner[k] == '\n') { ++k; }
-    if (inner[k] == 'F') {
+    if (inner[k] == 'O' || inner[k] == 'o') {
+      if (inner.substr(k, 8) == "OPTIONAL"
```

Comment:
Currently only all upper and all lower case are supported. The standard says most keywords are case insensitive (apart from the keyword 'a' which may only be lower case). It may be cleaner to change that in another pull request /. commit though.